libraries/Wire/examples/i2c_scanner/i2c_scanner.ino

```diff
@@ -29,15 +31,29 @@
 
 #include <Wire.h>
 
+uint16_t wireTimeoutCount; //capture I2C bus timeout events
+
 void setup() {
   Wire.begin();
 
   Serial.begin(9600);
   while (!Serial); // Leonardo: wait for serial monitor
   Serial.println("\nI2C Scanner");
+
+  Wire.setWireTimeout(3000, true); //timeout value in uSec, true to reset I2C bus on timeout
+  wireTimeoutCount = 0;
+  Wire.clearWireTimeoutFlag();
```

Comment:
Why not put this directly after the `Wire.begin()`? Seems more clear? Also, is clearing the timeout flag really needed here? We can rely on it being cleared on startup, I think? Same for initializing the count to 0, why not just do that in the variable declaration?

Reply:
Thanks for the comment; I moved the anti-lockup code to just under 'Wire.begin()' as you suggested.  I wanted to put the timeout flag in to expose its existence to a new user; otherwise they might never know the timeout flag existed at all or how to query it.

I could initialize the variable  in the declaration, but I have learned that assuming a variable has been initialized to zero 'somewhere else' can lead to confusing/incorrect behavior.   I like making things obvious and explicit, even at the cost of an extra line or two of code.  YMMV ;-).

Reply:
>  I wanted to put the timeout flag in to expose its existence to a new user; otherwise they might never know the timeout flag existed at all or how to query it.

Yeah, that's fine (and I think this might be a good example for it), I was just suggesting not doing the timeout flag handling in *all* examples (OTOH, maybe it should be).

> I like making things obvious and explicit, even at the cost of an extra line or two of code. YMMV ;-).

It does (on this particular thing, I do like explicit in general, but also concise), but it's ok :-)

What about (not) calling `clearWireTimeoutFlag()` on startup? You've now added a comment that seems less than helpful?

Reply:
Good catch!  I removed the call to clearWireTimeoutFlag() as it is explicitly cleared in Wire.setWireTimeout()